### PR TITLE
Replace inefficient uses of Process

### PR DESCRIPTION
**Pull request by K-Cully**
In some logging classes Process.GetCurrentProcess is used incorrectly.
This change fixes these use cases.

- Replace Process.GetCurrentProcess.Id in initialization logger, with Environment.ProcessId to prevent unnecessary Process initialization. (https://docs.microsoft.com/en-us/dotnet/api/system.environment.processid?view=net-6.0).
- Add using scope around Process in OmexLogEventSender to ensure it is disposed correctly.

Fixes https://github.com/microsoft/Omex/issues/453

### CODE REVIEW

File: src/Logging/InitializationLogger.cs [hunk 1]
```diff
@@ -41,7 +41,7 @@ private static ILoggingBuilder LoadInitializationLogger(this ILoggingBuilder bui
 		public static void LogInitializationSucceed(string serviceNameForLogging, string message = "")
 		{
 			string logMessage = $"Initialization successful for {serviceNameForLogging}, {message}";
-			ServiceInitializationEventSource.Instance.LogHostBuildSucceeded(Process.GetCurrentProcess().Id, serviceNameForLogging, logMessage);
+			ServiceInitializationEventSource.Instance.LogHostBuildSucceeded(Environment.ProcessId, serviceNameForLogging, logMessage);
```
Comment (K-Cully) [inline src/Logging/InitializationLogger.cs:44]: Environment.ProcessId doesn't exist in NetStandard 2.0

Comment (dimeehan) [inline src/Logging/InitializationLogger.cs:44]: Is Environment.ProcessId correct? I see some failures below saying there is no definition for 'ProcessId' ?

Comment (K-Cully) [reply src/Logging/InitializationLogger.cs:44]: @artempushkin - is there any reason the Microsoft.Omex.Extensions.Logging should support NetStandard 2.0?
I'd assume at this stage we wouldn't want to support FullFramework in downstream consumers, right?

File: src/Logging/Microsoft.Omex.Extensions.Logging.csproj [hunk 2]
```diff
@@ -2,6 +2,9 @@
   <PropertyGroup>
     <TargetFrameworks>$(LibraryTargetFrameworks)</TargetFrameworks>
   </PropertyGroup>
+  <PropertyGroup Condition="$(TargetFramework.StartsWith(`netstandard`))">
```
Comment (AndreyTretyak) [inline src/Logging/Microsoft.Omex.Extensions.Logging.csproj:5]: MSBuild defines the default constant `NETSTANDARD` no need to create another one https://docs.microsoft.com/en-us/dotnet/standard/frameworks#how-to-specify-a-target-framework

File: src/Logging/InitializationLogger.cs (hunk 1 above)
Comment (artempushkin) [reply src/Logging/InitializationLogger.cs:48]: We do target netframework in a few services, mostly in OmexAddInCertification. I suppose there can be partner dependencies forcing full .net.

Comment (K-Cully) [reply src/Logging/InitializationLogger.cs:48]: I went ahead with using conditional compilation, but I feel like we should move away from net standard in future.

Comment (K-Cully) [reply src/Logging/InitializationLogger.cs:48]: Yes, as per my comment above Environment.ProcessId doesn't exist in NetStandard 2.0.
It was only added in Net5. (https://docs.microsoft.com/en-us/dotnet/api/system.environment.processid?view=net-6.0)
The main benefits are that it:
- Caches the ID for any future calls.
- Prevents creation of a new process, just to get the ID.
- Avoids memory leaks due to developers forgetting to dispose of the Process instance that is created by GetCurrentProcess()

To get around the Net5+ requirement, I've specified conditional inclusion of this logic, and wrapped the NetStandard code in a using, to ensure the process gets disposed.

File: src/Logging/Microsoft.Omex.Extensions.Logging.csproj (hunk 2 above)
Comment (K-Cully) [reply src/Logging/Microsoft.Omex.Extensions.Logging.csproj]: Thanks Andrey I was looking for something like that but didn't come across that doc. :)

File: src/Logging/InitializationLogger.cs (hunk 1 above)
Comment (K-Cully) [reply src/Logging/InitializationLogger.cs:45]: Changed the variables used based on the doc Andrey linked.
I think the use of NET5_0_OR_GREATER adds more context.

**Review by neilr81 — APPROVED** [review]

**Pull request merged**